### PR TITLE
Add plugin for using hesim inline

### PR DESCRIPTION
**Pull request by mclements**
Devin: can you check this addition, please? It should allow for using `hesim` inline. Kindly, Mark.

### CODE REVIEW

**Comment by mclements** [comment]
I have now made the inline plugin internal -- which addresses an issue with .pkgdown.

**Comment by mclements** [comment]
Closing and re-opening to re-check:)

Is there another way to do this?

File: R/utils.R [hunk 1]
```diff
@@ -377,3 +377,33 @@ get_n_samples.array <- function(x){
...
+#' // [[Rcpp::export]]
+#' double test_inline_gengamma(double mu, double sigma, double Q) {
+#'   hesim::stats::gengamma gg(mu, sigma, Q);
+#'   return gg.random();
+#' }")
+#' set.seed(12345)
+#' test_inline_gengamma(1.0, 1.0, 1.0)
+#' @keywords internal
+#' @rdname plugin
+inlineCxxPlugin <- function(...) {
```
Comment (dincerti) [inline R/utils.R:398]: Lets stick to `camel_case` for functions:

```suggestion
inline_cxx_plugin <- function(...) {
```

File: R/utils.R [hunk 2]
```diff
@@ -377,3 +377,33 @@ get_n_samples.array <- function(x){
...
+#' Code to use the hesim package inline. Not directly called by the user.
+#' @param ... arguments
+#' @examples
+#' library(Rcpp)
+#' sourceCpp(code="
+#' // [[Rcpp::depends(hesim)]]
+#' // [[Rcpp::depends(RcppArmadillo)]]
+#' #include <hesim/stats/distributions.h>
+#' // [[Rcpp::export]]
+#' double test_inline_gengamma(double mu, double sigma, double Q) {
```
Comment (dincerti) [inline R/utils.R:390]: It would be good to add something like this as a unit test.

**Review by dincerti — CHANGES_REQUESTED** [review]
Thanks for the PR @mclements. Just a couple of minor comments. Also just curious: what is your intended use case for this?

File: R/utils.R (hunk 1 above)
Comment (mclements) [reply R/utils.R:398]: Devin: the name inlineCxxPlugin is required.

File: R/utils.R (hunk 2 above)
Comment (dincerti) [reply R/utils.R]: Looks like you had to remove the test?

Comment (mclements) [reply R/utils.R]: Yes: testing inline seems to require an installed package (using package/include/...) and does not work with a source tree (using package/inst/include/...). There may be some test magic to solve this. The test is also slow (2-4 seconds).

Note that I have found the inline facility very helpful for prototyping the cCTSTM implementations.

**Review by dincerti — APPROVED** [review]

**Pull request merged**